src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -230,13 +230,24 @@ void AddItemsToBatch(IEnumerable<TItem> items)
                 if (nextBatch.IsEmpty)
                     return default;
 
-                return await _processBatchAsync(nextBatch, batchCancellationToken).ConfigureAwait(false);
-            }
-            catch (OperationCanceledException) when (!_entireQueueCancellationToken.IsCancellationRequested)
-            {
-                // Don't bubble up cancellation to the queue for the nested batch cancellation.  Just because we decided
-                // to cancel this batch isn't something that should stop processing further batches.
-                return default;
+                var batchResultTask = _processBatchAsync(nextBatch, batchCancellationToken).Preserve();
+                await batchResultTask.NoThrowAwaitableInternal(false);
```

Comment:
📝 This change is retained because it does not overlap with compiler work. The no-throw logic in async state machines does not apply to `await` statements within a `try` block that contains one or more `catch` clauses, so avoiding an exception here requires updating to a no-throw form of awaiter.